### PR TITLE
Client Requirements

### PR DESCRIPTION
**Pull request by showierdata9978**
Continuation of https://github.com/meower-media-co/RFC/pull/2 

Clients that agree:
- [x] Roarer
- [x] meo

### CODE REVIEW

**Review by JoshAtticus — CHANGES_REQUESTED** [review]
This should not be markdown, it should be accessible from the actual website.

**Comment by williamhorning** [comment]
I disagree with Josh here since this is for contributing to the website repo. This isn't really content for the website

**Comment by showierdata9978** [comment]
> This should not be markdown, it should be accessible from the actual website.

invalid, this is not for the website itself

**Comment by JoshAtticus** [comment]
> > This should not be markdown, it should be accessible from the actual website.
> 
> invalid, this is not for the website itself

Then why in website repo?

**Comment by JoshAtticus** [comment]
> I disagree with Josh here since this is for contributing to the website repo. This isn't really content for the website

You disagree but agree?

**Comment by showierdata9978** [comment]
> > > This should not be markdown, it should be accessible from the actual website.
> > 
> > 
> > invalid, this is not for the website itself
> 
> Then why in website repo?

this is... a CONTRIBUTING.md...

**Comment by JoshAtticus** [comment]
> > > > This should not be markdown, it should be accessible from the actual website.
> > > 
> > > 
> > > invalid, this is not for the website itself
> > 
> > 
> > Then why in website repo?
> 
> this is... a CONTRIBUTING.md...

But it should still be accessible from website if people want to be able to easily figure out how to add their client to featured clients.

**Review by JoshAtticus — DISMISSED** [review]
Should be accessible from website in some way

**Comment by showierdata9978** [comment]
> Should be accessible from website in some way

not applicable to this PR

**Review by JoshAtticus — DISMISSED** [review]
Should still be accessible from website in some way even if it's a link to GitHub. Furthermore a CONTRIBUTING.md file only makes sense in the Website repository if it is for contributing to the website.

**Comment by showierdata9978** [comment]
> Should still be accessible from website in some way even if it's a link to GitHub. Furthermore a CONTRIBUTING.md file only makes sense in the Website repository if it is for contributing to the website.

THERE IS A LINK TO THE GH ON THE SITE
image

**Comment by JoshAtticus** [comment]
> > Should still be accessible from website in some way even if it's a link to GitHub. Furthermore a CONTRIBUTING.md file only makes sense in the Website repository if it is for contributing to the website.
> 
> THERE IS A LINK TO THE GH ON THE SITE image

Is that a link to the CONTRIBUTING.md file that isn't for contributing to the website?

**Comment by showierdata9978** [comment]
see https://discord.com/channels/1085009166894108673/1125397485532225637/1262236001846427689

**Comment by JoshAtticus** [comment]
> see https://discord.com/channels/1085009166894108673/1125397485532225637/1262236001846427689 (team only)

It should not be called CONTRIBUTING.md in the root directory of the website repository unless it's actually for contributing to the website.

**Comment by JoshAtticus** [comment]
> > see https://discord.com/channels/1085009166894108673/1125397485532225637/1262236001846427689 (team only)
> 
> It should not be called CONTRIBUTING.md in the root directory of the website repository unless it's actually for contributing to the website.

Would like to publically archive Showier's unprofessional response.

Screenshot 2024-07-15 at 11 12 27 am

**Comment by showierdata9978** [comment]
> > > see [discord.com/channels/1085009166894108673/1125397485532225637/1262236001846427689](https://discord.com/channels/1085009166894108673/1125397485532225637/1262236001846427689) (team only)
> > 
> > 
> > It should not be called CONTRIBUTING.md in the root directory of the website repository unless it's actually for contributing to the website.
> 
> Would like to publically archive Showier's unprofessional response.
> Screenshot 2024-07-15 at 11 12 27 am

look thats not changing because it was already forced to be changed.

**Comment by JoshAtticus** [comment]
> > > see [discord.com/channels/1085009166894108673/1125397485532225637/1262236001846427689](https://discord.com/channels/1085009166894108673/1125397485532225637/1262236001846427689) (team only)
> > 
> > 
> > It should not be called CONTRIBUTING.md in the root directory of the website repository unless it's actually for contributing to the website.
> 
> look thats not changing because it was already forced to be changed.

William isn't in control of everything, you don't have to do what he says. I am merely suggesting to change the file name or move it somewhere else and at least provide a direct link to the file on the website.

**Comment by showierdata9978** [comment]
> > > > see [discord.com/channels/1085009166894108673/1125397485532225637/1262236001846427689](https://discord.com/channels/1085009166894108673/1125397485532225637/1262236001846427689) (team only)
> > > 
> > > 
> > > It should not be called CONTRIBUTING.md in the root directory of the website repository unless it's actually for contributing to the website.
> > 
> > 
> > look thats not changing because it was already forced to be changed.
> 
> William isn't in control of everything, you don't have to do what he says. I am merely suggesting to change the file name or move it somewhere else and at least provide a direct link to the file on the website.

tbf you also quoted it.

**Comment by JoshAtticus** [comment]
> > > > > see [discord.com/channels/1085009166894108673/1125397485532225637/1262236001846427689](https://discord.com/channels/1085009166894108673/1125397485532225637/1262236001846427689) (team only)
> > > > 
> > > > 
> > > > It should not be called CONTRIBUTING.md in the root directory of the website repository unless it's actually for contributing to the website.
> > > 
> > > 
> > > look thats not changing because it was already forced to be changed.
> > 
> > 
> > William isn't in control of everything, you don't have to do what he says. I am merely suggesting to change the file name or move it somewhere else and at least provide a direct link to the file on the website.
> 
> tbf you also quoted it.

Where?

**Comment by showierdata9978** [comment]
> snip
> > tbf you also quoted it.
> 
> Where?
https://github.com/meower-media-co/RFC/pull/2#pullrequestreview-2176780826

**Comment by mybearworld** [comment]
This is for contributing to the website in that you're trying to get a client featured on the website.

---

Is account management stuff like changing your password, deleting your account and revoking tokens (even though the latter is pretty much obsolete now) intentionally left out?

**Comment by JoshAtticus** [comment]
> This is for contributing to the website in that you're trying to get a client featured on the website.

CONTRIBUTING.md (to my understanding) is for how to contribute to a repo. Rules for featured client should not be the only thing in CONTRIBUTING.md (or in it at all in my opinion), it should be in it's own page or file accessible from the website.

**Comment by JoshAtticus** [comment]
> > snip
> > > tbf you also quoted it.
> > 
> > 
> > Where?
> > [meower-media-co/RFC#2 (review)](https://github.com/meower-media-co/RFC/pull/2#pullrequestreview-2176780826)

He never said to call it CONTRIBUTING.md in any of those and it's still a good idea to make it accessible from the website.

**Comment by mybearworld** [comment]
> CONTRIBUTING.md (to my understanding) is for how to contribute to a repo. Rules for featured client should not be the only thing in CONTRIBUTING.md (or in it at all in my opinion), it should be in it's own page or file accessible from the website.

Yeah you're right

**Comment by showierdata9978** [comment]
there is nothing else rn for the file. So its a top level... I can make it secondary level

**Comment by mybearworld** [comment]
It should probably be a part of the website itself as that makes it much more easily accessible, and easier to know about in the first place.

**Comment by JoshAtticus** [comment]
> It should probably be a part of the website itself as that makes it much more easily accessible, and easier to know about in the first place.

Exactly my point

**Comment by 3r1s-s** [comment]
personally i think this would be nice to have on a page here

**Comment by showierdata9978** [comment]
*oh my god I said that is out of scope for this PR*

**Comment by 3r1s-s** [comment]
too bad didnt read

File: Clients.md
Comment (tnix100) [inline Clients.md:52]: This should be optional. Meower Uploads exist and the majority of users will use it rather than linking to images nowadays.

File: Clients.md
Comment (tnix100) [inline Clients.md:32]: What about updating chats when there's an update_chat event?

File: Clients.md
Comment (tnix100) [inline Clients.md:34]: Why is this indented but profile picture support isn't?

File: Clients.md
Comment (tnix100) [inline Clients.md:46]: This is indented under Settings, doesn't make a lot of sense.

File: Clients.md
Comment (tnix100) [inline Clients.md:10]: Why? If they don't talk to anything but Meower, there's no need for an extra privacy policy.

File: Clients.md
Comment (tnix100) [inline Clients.md:21]: This should be future-proofed, and shouldn't be a requirement. Change it to something like "You should try to use the latest available API versions."

**Review by tnix100 — CHANGES_REQUESTED** [review]
Honestly, this and the RFC that this relates to needs some major improvements. Lack of detail and unnecessary requirements.

File: Clients.md
Comment (showierdata9978) [reply Clients.md:34]: pfps are seen globally

File: Clients.md
Comment (tnix100) [inline Clients.md:20]: Remove "must", shouldn't be a requirement. There is also many APIs, and technically the WebSocket could be seen as an API, so I'd just make it more generic to cover every API.

Comment (showierdata9978) [reply Clients.md:20]: aaaand github did not include the commit suggestion thing

Comment (showierdata9978) [reply Clients.md:20]: oh wait
that was for the specific commit nvn

**Review by JoshAtticus — CHANGES_REQUESTED** [review]
Create page on website containing the rules or at least link to it from the website, it's not even that hard it will take you a whole 1 minute

**Review by JoshAtticus — CHANGES_REQUESTED** [review]
Create page on website containing the rules or at least link to it from the website, it's not even that hard it will take you a whole 1 minute

**Comment by showierdata9978** [comment]
> Create page on website containing the rules or at least link to it from the website, it's not even that hard it will take you a whole 1 minute

That is NOT happening in this PR. stop spamming

**Comment by JoshAtticus** [comment]
> > Create page on website containing the rules or at least link to it from the website, it's not even that hard it will take you a whole 1 minute
> 
> That is NOT happening in this PR. stop spamming

I'm not spamming it's literally not even that hard to add a link and if you don't I'll do it myself

**Review by JoshAtticus — CHANGES_REQUESTED** [review]
Add link to file on website to show its existence

**Review by JoshAtticus — CHANGES_REQUESTED** [review]
For myself, add link to file on website to show its existence. Dismiss and I will re-add

**Comment by 3r1s-s** [comment]
> Create page on website containing the rules or at least link to it from the website, it's not even that hard it will take you a whole 1 minute

yea ill do that once the thing is a completed state :yuhhuh:

**Review by JoshAtticus — DISMISSED** [review]
For myself, add link to file on website to show its existence. Dismiss and I will re-add

**Review by 3r1s-s — COMMENTED** [review]
yes

**Comment by JoshAtticus** [comment]
Showier stop hiding comments

**Comment by showierdata9978** [comment]
Josh it was CLIENTS.md on purpose

**Comment by JoshAtticus** [comment]
> Josh it was CLIENTS.md on purpose

then make it all caps not first letter caps that looks ugly

**Comment by showierdata9978** [comment]
> > Josh it was CLIENTS.md on purpose
> 
> then make it all caps not first letter caps that looks ugly

then thats a gh issue, It was CLIENTS.md when I did it

**Comment by JoshAtticus** [comment]
> > > Josh it was CLIENTS.md on purpose
> > 
> > 
> > then make it all caps not first letter caps that looks ugly
> 
> then thats a gh issue, It was CLIENTS.md when I did it

are you gaslighting or what I did CLIENTS with github web and it worked? never had git change it to only first letter capital before either

**Comment by tnix100** [comment]
This appears to just be for the MD file and doesn't need to be added to the site yet. I'm fine with this PR being merged after the MD file is finished.

**Comment by tnix100** [comment]
But, should be added to the site ASAP after this PR gets merged.

**Comment by showierdata9978** [comment]
> > > > Josh it was CLIENTS.md on purpose
> > > 
> > > 
> > > then make it all caps not first letter caps that looks ugly
> > 
> > 
> > then thats a gh issue, It was CLIENTS.md when I did it
> 
> are you gaslighting or what I did CLIENTS with github web and it worked? never had git change it to only first letter capital before either
I had this issue with git on mb.py. It did not change the casing.

**Comment by 3r1s-s** [comment]
> But, should be added to the site ASAP after this PR gets merged.

noted

**Comment by mybearworld** [comment]
Reactions/replies?

**Comment by showierdata9978** [comment]
> Reactions/replies?

oh right they where only recently added so I forgot about them

**Review by commandc904 — APPROVED** [review]

**Review by MikeDev101 — APPROVED** [review]
I'm just clearing out my inbox. Refer to other member's reviews.

**Review by williamhorning — APPROVED** [review]

**Comment by showierdata9978** [comment]
@mybearworld @tnix100 your the last ppl blocking this from getting merged now

File: CLIENTS.md
Comment (mybearworld) [inline CLIENTS.md:31]: This should probably be more generic (or be implied), `update_chat` is not the only CL command

File: CLIENTS.md
Comment (mybearworld) [inline CLIENTS.md:47]: For this specifically, having documentation on how to do this is very important.

(It does makes sense as a requirement)

**Review by mybearworld — CHANGES_REQUESTED** [review]

File: CLIENTS.md
Comment (showierdata9978) [reply CLIENTS.md:47]: Im not sure how to include the documentation in here.

**Review by mybearworld — APPROVED** [review]

File: CLIENTS.md
Comment (williamhorning) [reply CLIENTS.md:47]: having documentation on this is a good idea, but it should be done separately from this

Comment (showierdata9978) [reply CLIENTS.md:47]: that should be on https://github.com/meower-media/docs

File: CLIENTS.md
Comment (JoshAtticus) [inline CLIENTS.md:24]: We should define "reasonable amount of time"

**Review by JoshAtticus — CHANGES_REQUESTED** [review]

File: CLIENTS.md
Comment (mybearworld) [inline CLIENTS.md:10]: This contradicts itself
```suggestion
- Clients must be open source
```

**Review by mybearworld — CHANGES_REQUESTED** [review]

File: CLIENTS.md
Comment (mybearworld) [inline CLIENTS.md:11]:
```suggestion
- Clients must be Open Source Software (Any OSI approved licence).
```

File: imageHosts.json
Comment (mybearworld) [inline imageHosts.json:1]: This really shouldn't be on meower.org

**Review by mybearworld — CHANGES_REQUESTED** [review]

File: CLIENTS.md
Comment (tehbarney86) [inline CLIENTS.md:57]: undocumented. (what time does an indicator signal last for?)

File: CLIENTS.md
Comment (tehbarney86) [inline CLIENTS.md:51]: undocumented. (still have no comprehension how to properly post a reaction, api.meower.org/docs doesnt help at all)

File: CLIENTS.md
Comment (tehbarney86) [reply CLIENTS.md:62]: why? i dont think this list is getting bigger anytime soon anyways.

File: imageHosts.json
Comment (showierdata9978) [reply imageHosts.json:1]: Thats why it was not an api initially

File: CLIENTS.md
Comment (tehbarney86) [inline CLIENTS.md:36]: custom profile support? oh, the feature that svelte doesn't have for like 2 months already? (sorry for being passive-agressive, but this is just hypocrisy at this point)

custom profiles shouldn't be a requirement, like in meower95, they would break any sort of aesthetic/feel to it, especially with the 16x16 size. it also isn't easily supported everywhere, especially with most of them being webps.

y'all know not every client is web, right?

**Review by tehbarney86 — CHANGES_REQUESTED** [review]
seriously, a requirement list for clients should be about security and not just a list of features that can be just basically not supported by some clients. developing a client not online with all these requirements in is basically impossible with some languages.

File: CLIENTS.md
Comment (showierdata9978) [reply CLIENTS.md:58]: Use a prebuilt one then

File: CLIENTS.md
Comment (showierdata9978) [reply CLIENTS.md:44]: no.

File: CLIENTS.md
Comment (showierdata9978) [reply CLIENTS.md:57]: documentation is out of scope

File: CLIENTS.md
Comment (showierdata9978) [reply CLIENTS.md:51]: documentation is out of scope

File: CLIENTS.md
Comment (showierdata9978) [reply CLIENTS.md:36]: > custom profile support? oh, the feature that svelte doesn't have for like 2 months already? (sorry for being passive-agressive, but this is just hypocrisy at this point)
> 
> custom profiles shouldn't be a requirement, like in meower95, they would break any sort of aesthetic/feel to it, especially with the 16x16 size. it also isn't easily supported everywhere, especially with most of them being webps.
> 
> y'all know not every client is web, right?

for 1, svelte is now just called client. 2 it is being worked on privately.

File: CLIENTS.md
Comment (showierdata9978) [reply CLIENTS.md]: thats already done on another line

File: CLIENTS.md
Comment (mybearworld) [reply CLIENTS.md:58]: You really shouldn't use Python/TKinter to make Meower clients. You can't use CAPTCHAs with that either.

File: CLIENTS.md
Comment (mybearworld) [reply CLIENTS.md:44]: Clients that are featured on the Meower Website should follow some base line, where signing up is definitely part of it. Security via CAPTCHA is also important. That necessarily leads to web clients being preferred

File: CLIENTS.md
Comment (mybearworld) [reply CLIENTS.md:57]: three seconds

File: imageHosts.json
Comment (mybearworld) [reply imageHosts.json:1]: Could an endpoint for this be PRd into server before this gets merged then? As is, this would not work at all well

File: CLIENTS.md
Comment (tehbarney86) [reply CLIENTS.md:51]: why is that?

why then make a requirements page if the developers don't even know how to use your tools? this is just confusing, why is making a documentation for other developers to actually understand your servers is less important than continuing making it more complex and harder to make them more confused and making a documentation harder in the end?

File: CLIENTS.md
Comment (tehbarney86) [reply CLIENTS.md:58]: > You really shouldn't use Python/TKinter to make Meower clients. You can't use CAPTCHAs with that either.

that is a recommendation, not a requirement, and should never be. people have the freedom to make anything and making clients web-only is just basically unnecessary and annoying gatekeeping.

> Use a prebuilt one then

not every programming language has a "prebuilt" markdown parser, that was the point, and gatekeeping clients to be web-based is unnecessary and should be avoided.

File: CLIENTS.md
Comment (tehbarney86) [reply CLIENTS.md:44]: > Clients that are featured on the Meower Website should follow some base line, where signing up is definitely part of it. Security via CAPTCHA is also important. That necessarily leads to web clients being preferred

still, this is unnecessary and better alternatives always exist, and i don't think completely removing the security after all is a problem - browsers automatically include any kind of captcha, so spamming isn't a problem for troublemakers, needless gatekeeping.

_Voice communication is not available for this account._

File: imageHosts.json
Comment (tehbarney86) [reply imageHosts.json:1]: > This really shouldn't be on meower.org – that would require yet another server URL to be able to be set by clients

agreed, there is no meower.org/docs, there is no meower.org/svelte, things should really be in the right places here.

File: CLIENTS.md
Comment (tehbarney86) [reply CLIENTS.md:36]: why are they "absolutely required"? again, gatekeeping for no absolute reason.
alright, the 34 preset avatars are an alright and simple requirement, even with 2 text characters you can represent them. but enforcing custom avatars just make zero sense. what about text based clients? what about non-web clients? what about clients that don't use custom avatars for an aesthetic reason?

File: CLIENTS.md
Comment (tehbarney86) [inline CLIENTS.md:43]: undocumented.

File: CLIENTS.md
Comment (tehbarney86) [inline CLIENTS.md:13]: why then make a requirements page?

File: CLIENTS.md
Comment (tehbarney86) [inline CLIENTS.md:14]: again, why then make requirements in the first place? half of all the clients featured don't comply with the guidelines with often one, two, or maybe half the whole RFC.

File: CLIENTS.md
Comment (tehbarney86) [inline CLIENTS.md:21]:
```suggestion
- Use websockets to get updates to cached information
```

File: CLIENTS.md
Comment (tehbarney86) [inline CLIENTS.md:20]: very vague, this basically allows clients like meower95 to store deleted messages, a feature multiple times said as "not complying with RFC".

File: CLIENTS.md
Comment (tehbarney86) [inline CLIENTS.md:10]: define "malicious intent", this is just funny.

**Review by tehbarney86 — CHANGES_REQUESTED** [review]
still, requirements should be just for security/design, not enforcing features to only ignore them after all.

File: CLIENTS.md
Comment (showierdata9978) [reply CLIENTS.md:26]: Security features will NEVER be removed.

**Comment by showierdata9978** [comment]
You can always host a local thing when signing up.

**Comment by showierdata9978** [comment]
> still, requirements should be just for security/design, not enforcing features to only ignore them after all.

This has not been merged, and no feature requirements will be removed.

**Comment by tnix100** [comment]
Maybe no feature requirements will be removed, but this whole thing is going to be gotten rid of. 

- There shouldn't be feature requirements to be featured (even custom PFPs shouldn't be a requirement, not every client will even be able to render images).
- Security should be common sense.
- Not violating Meower's Terms of Service & Privacy Policy should be common sense.
- It should just be up to the team to feature clients, and we should communicate with client devs individually to address any issues before being featured. We'll have an internal guide on what to check for when deciding whether to feature a client, but there shouldn't be many strict feature requirements.

File: CLIENTS.md
Comment (WlodekM) [reply CLIENTS.md:43]: copy code from meo :trol:

**Pull request closed without merging**
